lib/content-utils.ts

```diff
@@ -0,0 +1,53 @@
+import fs from 'fs/promises';
+import path from 'path';
+import matter from 'front-matter';
+import MarkdownIt from 'markdown-it';
+
+const md = new MarkdownIt();
+
+export async function readContentFile(filePath: string): Promise<{ data: Record<string, any>; content: string } | undefined> {
+  const fileContent = await fs.readFile(filePath, 'utf8');
+  const extension = path.extname(filePath).toLowerCase();
+
+  if (extension === '.md') {
+    const parsed = matter(fileContent);
+    return { data: parsed.attributes as Record<string, any> || {}, content: md.render(parsed.body) };
+  } else if (extension === '.yaml' || extension === '.yml') {
+    // dynamically import js-yaml to decrease bundle size and initial load times, 
+    // also can help isolate dependencies and prevent unexpected interactions between parts of code
+    // change if many yaml files and dependency issue isn't a problem
+    const yaml = (await import('js-yaml')).load(fileContent) as Record<string, any>;
+    return { data: yaml, content: '' };
+  }
+}
+
+export async function readContentFolder(contentFolder: string) {
+  const folderPath = path.join('content', contentFolder);
+  const filenames = await fs.readdir(folderPath);
+
+  return Promise.all(
+    filenames.map(async (filename) => {
+      const filePath = path.join(folderPath, filename);
+      const parsed = await readContentFile(filePath);
```

Comment:
So I think this is actually the fileContent? `parsed` gets used later on as if it's a filename with an extension. Which is correct?

Reply:
parsed is the filecontent... it's used a few lines down to check if it's undefined before checking extensions